cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,415 @@
+ * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowExistsFilterButtonFiltersCorrectField(
+  docTableRowNumber,
+  expandedDocumentRowNumber,
+  expectedQueryHitsWithoutFilter,
+  expectedQueryHitsAfterFilterApplied
+) {
+  getExpandedDocRowFieldName(docTableRowNumber, expandedDocumentRowNumber).then(
+    ($expandedDocumentRowField) => {
+      const filterFieldText = $expandedDocumentRowField.text();
+      getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber).within(() => {
+        cy.getElementByTestId('addExistsFilterButton').click();
+      });
+      // Verify full pill text
+      // globalFilterLabelValue gives the inner element, but we may want all the text in the filter pill
+      cy.getElementByTestId('globalFilterLabelValue', {
+        timeout: 10000,
+      })
+        .parent()
+        .should('have.text', filterFieldText + ': ' + 'exists');
+      cy.getElementByTestId('discoverQueryHits').should(
+        'have.text',
+        expectedQueryHitsAfterFilterApplied
+      );
+    }
+  );
+  cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+}
+
+/**
+ * Sends a new query via the query multiline editor.
+ * @param del true/false. true: Deletes character to the right of the cursor; false: Deletes character to the left of the cursor
+ * @see https://docs.cypress.io/api/commands/type#Arguments
+ */
+export function sendQueryOnMultilineEditor(query) {
+  const getMultilineEditor = () => {
+    cy.getElementByTestId('headerGlobalNav').click(); // remove syntax helper
+    return cy.get('.view-line');
+  };
+  // Clear default text on the editor by an alternative method, since
+  // cy.clear() won't work for some reason
+  getMultilineEditor()
+    .invoke('text')
+    .then(function ($content) {
+      const contentLen = $content.length + 1;
+      getMultilineEditor().type('a'); // make sure we're at the end of the string
+      getMultilineEditor().type('{backspace}'.repeat(contentLen));
+    });
+  // Type query
+  getMultilineEditor().type(query);
+  // Send query
+  cy.getElementByTestId('querySubmitButton').click();
+}
+
+/**
+ * Set the query editor language
+ * @param language Accepted values: 'DQL', 'Lucene', 'OpenSearch SQL', 'PPL'
+ */
+export function setQueryEditorLanguage(language) {
+  cy.getElementByTestId('headerGlobalNav').click(); // remove helper message
+  cy.getElementByTestId('queryEditorLanguageSelector').click();
+  cy.getElementByTestId('queryEditorLanguageOptions').find('button').contains(language).click();
+}
```

Comment:
This functionality already exists via `cy.setQueryLanguage()`. Can you try using that one and remove this one if it works?